### PR TITLE
[18073] Monitor Service & Monitor Service Listener API

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR adds the `MonitorService` and `MonitorServiceListener` API that will implement the main logic of the feature.

_Note: Built on top of #3483_

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- **N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [x] Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A**  New feature has been added to the `versions.md` file (if applicable).
- **N/A**  New feature has been documented/Current behavior is correctly described in the documentation. 
    
- **N/A**  Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/statistics/rtps/monitor-service/MonitorServiceListener.cpp
Comment (jsantiago-eProsima) [inline src/cpp/statistics/rtps/monitor-service/MonitorServiceListener.cpp:1]: Newline at EOF

**Comment by Mario-DL** [comment]
@richiprosima please test this

File: src/cpp/statistics/rtps/monitor-service/MonitorService.hpp [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+namespace fastdds {
+namespace statistics {
+namespace rtps {
+
+#ifdef FASTDDS_STATISTICS
+
+class MonitorService
+{
+    //! Dummy implementation
+    struct SimpleQueryable : public IStatusQueryable
```
Comment (jsantiago-eProsima) [inline src/cpp/statistics/rtps/monitor-service/MonitorService.hpp:48]: This is a `class` in the design.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Review by jsantiago-eProsima — APPROVED** [review]
LGTM

**Pull request merged**